### PR TITLE
ci: check-api skips grpc namespace

### PR DESCRIPTION
**Pull request by dbolduc**
This would have saved me a commit on a different PR.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13391)

### CODE REVIEW

File: ci/cloudbuild/builds/check-api.sh [hunk 1]
```diff
@@ -123,7 +123,7 @@ function check_abi() {
       # characters in the string "internal", and it should again be followed
       # by some other number indicating the length of the symbol within the
       # "internal" namespace. See: https://en.wikipedia.org/wiki/Name_mangling
-      -skip-internal-symbols "(8internal|_internal)\d"
+      -skip-internal-symbols "(8internal|_internal|4grpc)\d"
```
Comment (coryan) [inline ci/cloudbuild/builds/check-api.sh:126]: Should we add `absl` while we are at it?

**Review by coryan — APPROVED** [review]

File: ci/cloudbuild/builds/check-api.sh (hunk 1 above)
Comment (dbolduc) [reply ci/cloudbuild/builds/check-api.sh:126]: Part of me doesn't want to think too hard about what skipping `absl::optional<T>` means.

Comment (coryan) [reply ci/cloudbuild/builds/check-api.sh:126]: > Have you had an `absl` false positive before?

Yes.

> Part of me doesn't want to think too hard about what skipping `absl::optional<T>` means.

Fair enough.

Comment (alevenberg) [reply ci/cloudbuild/builds/check-api.sh:126]: I have an absl flake 

https://github.com/googleapis/google-cloud-cpp/pull/13596/checks?check_run_id=21537016008

Comment (devbww) [reply ci/cloudbuild/builds/check-api.sh:126]: That seems like a true finding ...
```
absl::forward<std::__cxx11::basic_string<char>const&> ( remove_reference_t& t )
```
changed to ...
```
absl::forward<std::__cxx11::basic_string<char>&> ( remove_reference_t& t )
```
Perhaps that is not an "API change" however (just losing `const`), and regenerating the ABI dump is OK?

Comment (devbww) [reply ci/cloudbuild/builds/check-api.sh:126]: Note that this is a template instantiation, so perhaps we lost and gained a symbol just by changing our own type?

**Pull request merged**